spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -501,6 +503,32 @@ public void testDeleteNonExistingRecords() {
         sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", selectTarget()));
   }
 
+  @TestTemplate
+  public void deleteSingleRecordProducesDeleteOperation() {
```

Comment:
something seems to be off with subclasses of `TestDelete`. The test produces different results on each run when executing `./gradlew :iceberg-spark:iceberg-spark-extensions-3.5_2.12:test --tests "org.apache.iceberg.spark.extensions.TestCopyOnWriteDelete" --tests "org.apache.iceberg.spark.extensions.TestMergeOnReadDelete"` vs when running `TestCopyOnWriteDelete` or `TestMergeOnReadDelete` within the IDE

Reply:
turns out this was because of the randomness of adaptive query execution: https://github.com/apache/iceberg/blob/d1e24f4c9571ab7fba66d692cde28188f265d34e/spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/ExtensionsTestBase.java#L56

Reply:
I'll port this test to Spark 3.3/3.4 once CI passes

Reply:
I may have an explanation [here](https://github.com/apache/iceberg/pull/10123#discussion_r1566275123).